### ISSUE TITLE
TS Lint Server crashes when tslint.nodePath does not exist

### ISSUE THREAD

**Issue by garthk** [Microsoft/vscode-tslint#376]
- tslint Version: 5.10.0
- typescript Version: 2.9.1
- tslint extension 1.0.34

Steps to Reproduce:

0. `npm install` to get your dev-dependency on `tslint`
0. Set `tslint.nodePath` to `"${workspaceFolder}/node_modules"` in your workspace settings
0. **Re-launch VS Code 1.26.0 after update**
0. Navigate to some TypeScript
0. Observe error notification: 

      > ⊗ The tslint server crashed 5 times in the last 3 minutes. The server will not be restarted.

0. ⌘⇧P `TSLint: Show output`
0. Observe:

        [Info  - 9:05:26 AM] Linter is running.
        [Info  - 9:05:26 AM] Connection to server got closed. Server will restart.
        [Info  - 9:05:26 AM] Linter has stopped.
        [Info  - 9:05:27 AM] Linter is running.
        [Info  - 9:05:27 AM] Connection to server got closed. Server will restart.
        [Info  - 9:05:27 AM] Linter has stopped.
        [Info  - 9:05:28 AM] Linter is running.
        [Info  - 9:05:28 AM] Connection to server got closed. Server will restart.
        [Info  - 9:05:28 AM] Linter has stopped.
        [Info  - 9:05:29 AM] Linter is running.
        [Info  - 9:05:29 AM] Connection to server got closed. Server will restart.
        [Info  - 9:05:29 AM] Linter has stopped.
        [Info  - 9:05:30 AM] Linter is running.
        [Error - 9:05:30 AM] Connection to server got closed. Server will not be restarted.
        [Info  - 9:05:30 AM] Linter has stopped.
      
0. Set `tslint.nodePath` to `"${workspaceRoot}/node_modules"` in your workspace settings
0. ⌘⇧P `Reload`
0. Observe errors as above
0. Remove `tslint.nodePath` from your workspace settings
0. Confirm there's no `tslint.nodePath` in your user settings
0. ⌘⇧P `Reload`
0. Observe listing now works with the global `tslint` 5.10.0
0. From the command line, `cd` to the workspace root and:

        $ node_modules/.bin/tslint 'src/**/*.ts'

0. Observe linting works
0. From the command line:

        $ node
        > require('tslint')

0. Observe you can `require` the module just fine

The workspace setting reaching into `node_modules` to pick up the `tslint` version referred to in `devDependencies` used to work. Did something change?

**Comment by egamma** [comment]
With 1.26 there comes a new electron version with a new node version and vscode-tslint is using the same node version as vscode.

However, I´m not able to reproduce this on windows. This is the setup I´ve used

- create an empty folder
- npm install tslint typescript
- setup tslint.json and define `tslint.nodePath` as shown in the screenshot below

![image](https://user-images.githubusercontent.com/172399/44083649-ea553470-9fb4-11e8-9b69-af8266a58926.png)

Is anything different in your setup?

Can you start the server with verbose tracing on (see https://github.com/Microsoft/vscode-tslint/blob/master/tslint/README.md#trouble-shooting) and attach the trace.

**Comment by timothyallan** [comment]
I'm getting this as well after updating to 1.26. On any project, even brand new ones.

Only tslint / typescript installed:

![screen shot 2018-08-15 at 11 50 52 am](https://user-images.githubusercontent.com/1995135/44163783-8a850980-a081-11e8-9d39-973c5b18e4eb.png)




MacOS 10.13.5
VSCode 1.26
TSLint 1.0.34

```
[Info  - 11:37:49 AM] Linter is running.
[Info  - 11:37:49 AM] Connection to server got closed. Server will restart.
[Info  - 11:37:49 AM] Linter has stopped.
[Info  - 11:37:49 AM] Linter is running.
[Info  - 11:37:50 AM] Connection to server got closed. Server will restart.
[Info  - 11:37:50 AM] Linter has stopped.
[Info  - 11:37:50 AM] Linter is running.
[Info  - 11:37:50 AM] Connection to server got closed. Server will restart.
[Info  - 11:37:50 AM] Linter has stopped.
[Info  - 11:37:50 AM] Linter is running.
[Info  - 11:37:50 AM] Connection to server got closed. Server will restart.
[Info  - 11:37:50 AM] Linter has stopped.
[Info  - 11:37:51 AM] Linter is running.
[Error - 11:37:51 AM] Connection to server got closed. Server will not be restarted.
[Info  - 11:37:51 AM] Linter has stopped.
```

**Comment by egamma** [comment]
@timothyallan can you please run this with verbose tracing enabled and add the trace. To enable more tracing add the setting "tslint.trace.server" with a value of "verbose".

I also look into whether I can repro this on a OS X.

**Comment by egamma** [comment]
Update: I can reproduce the problem on OS X when the tslint.nodePath is not correct.

**Comment by egamma** [comment]
published fix in 1.0.35
